tutorial/function-pointers-as-arguments.md

```diff
@@ -0,0 +1,250 @@
+    y = function2(a);
+    return x + y;
+}
+/*........................................*/
+```
+
+Here is the Fortran equivalent:
+
+```Fortran
+module boo
+
+
+  use, intrinsic :: ISO_C_Binding, only : i64 => C_INT64_T
+  implicit none
+
+  contains
+
+  !........................................
+  function high_int_int_1(function1, function2, a) result(Out_0001)
+
+    implicit none
+
+    integer(i64) :: Out_0001
+    integer(i64), value :: a
+    integer(i64) :: x
+    integer(i64) :: y
+
+    interface
+      function function1(in_0000) result(out_0000)
+        use, intrinsic :: ISO_C_Binding, only : i64 => C_INT64_T
+        integer(i64) :: out_0000
+        integer(i64), value :: in_0000
+      end function function1
+
+      function function2(in_0001) result(out_0001)
+        use, intrinsic :: ISO_C_Binding, only : i64 => C_INT64_T
+        integer(i64) :: out_0001
+        integer(i64), value :: in_0001
+      end function function2
+    end interface
+
+    x = function1(a)
+    y = function2(a)
+    Out_0001 = x + y
+    return
+
+  end function high_int_int_1
+  !........................................
+
+end module boo
+```
+
+Here is the version using a function-header:
+
+```Python
+#$ header function high_int_int_1((int)(int), (int)(int), int)
+def high_int_int_1(function1, function2, a):
+    x = function1(a)
+    y = function2(a)
+    return x + y
+```
+
+Here is the generated C code:
+
+```C
+#include "boo.h"
+#include <stdlib.h>
+#include <stdint.h>
+
+
+/*........................................*/
+int64_t high_int_int_1(int64_t (*function1)(int64_t ), int64_t (*function2)(int64_t ), int64_t a)
+{
+    int64_t x;
+    int64_t y;
+    x = function1(a);
+    y = function2(a);
+    return x + y;
+}
+/*........................................*/
+```
+
+And the Fortran equivalent:
+
+```Fortran
+module boo
+
+
+  use, intrinsic :: ISO_C_Binding, only : i64 => C_INT64_T
+  implicit none
+
+  contains
+
+  !........................................
+  function high_int_int_1(function1, function2, a) result(Out_0001)
+
+    implicit none
+
+    integer(i64) :: Out_0001
+    integer(i64), value :: a
+    integer(i64) :: x
+    integer(i64) :: y
+
+    interface
+      function function1(in_0000) result(out_0000)
+        use, intrinsic :: ISO_C_Binding, only : i64 => C_INT64_T
+        integer(i64) :: out_0000
+        integer(i64), value :: in_0000
+      end function function1
+
+      function function2(in_0001) result(out_0001)
+        use, intrinsic :: ISO_C_Binding, only : i64 => C_INT64_T
+        integer(i64) :: out_0001
+        integer(i64), value :: in_0001
+      end function function2
+    end interface
+
+    x = function1(a)
+    y = function2(a)
+    Out_0001 = x + y
+    return
+
+  end function high_int_int_1
+  !........................................
+
+end module boo
+```
+
+Now, We will see an advnaced example of this feature with `const`  keyword and using multiple arguments function-type:
```

Comment:
I think it is important to explain why we care about the `const` keyword for function-pointers more than elsewhere. Do you understand why this is the case or do you want me to write something?